Assets/Scripts/Drone Scripts/Drone.cs

```diff
@@ -4,16 +4,20 @@
     using System.Collections.Generic;
     using UnityEngine;
     using ROSBridgeLib.interface_msgs;
+    using UnityEngine.UI;
 
     public class Drone
     {
 
         public GameObject gameObjectPointer; // This is the related game object
+        public DroneProperties droneProperties;
         public char id; // This is the identifier of the drone in the dronesDict and across the ROSBridge
         public bool selected;
 
         public ArrayList waypoints; // All waypoints held by the drone
```

Comment:
Maybe its a C# thing but whats the type of the array list? Does it hold ints? Waypoints?

Reply:
@apollovision

Reply:
Holds waypoint objects, which themselves hold coordinates and various graphics elements.

Reply:
I think the arrayList can hold anything, but since you mentioned it, might be a good idea to instead use an array of waypoints instead of an arraylist.

Reply:
resolved by using a simple list instead, which is all that is required